File: ObamaClient/src/app/model/group.ts
import {User} from './user';

export class Group {

  private _id: number;
  private _groupName: String;
  private _group_creator: number;
  private _group_topic: String;

  constructor(id: number, group_name: String, group_creator: number, group_topic: String) {
    this._id = id;
    this._groupName = group_name;
    this._group_creator = group_creator;
    this._group_topic = group_topic;
  }


  get id(): number {
    return this._id;
  }

  get groupName(): String {
    return this._groupName;
  }

  get group_creator(): number {
    return this._group_creator;
  }

  get group_topic(): String {
    return this._group_topic;
  }
}

File: graubunden/src/main/java/nu/obama/graubunden/model/Group.java
package nu.obama.graubunden.model;

import org.hibernate.annotations.NaturalId;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;

@Entity
@Table(name = "\"groups\"")
public class Group {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    // Group shall contain the total amount of upboats from all the posts upboats.
    // All groups
    @NotBlank
    @Column(length = 30)
    private String groupName;

    @ManyToOne
    @JoinColumn(name = "users")
    private User user;

    @NotBlank
    @Column(length = 30)
    private String groupTopic;

    public Group(){

    }

    public Group(String groupName, String groupTopic) {
        this.groupName = groupName;
        this.groupTopic = groupTopic;
    }
    public void setUser(User user) {
        this.user = user;
    }
    public void getUser(User user) {
        this.user = user;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getGroupName() {
        return groupName;
    }

    public void setGroupName(String groupName) {
        this.groupName = groupName;
    }

    public String getGroupTopic() {
        return groupTopic;
    }

    public void setGroupTopic(String groupTopic) {
        this.groupTopic = groupTopic;
    }

}
